tests/dispatch/test_scheduler.py

```diff
@@ -116,10 +123,15 @@ async def main():
             )
 
         output = self.start(main)
-
+        # In this test, the output is deterministic, but it does not follow the
+        # order in which the coroutines are declared due to interleaving of the
+        # asyncio event loop.
+        #
+        # Note that the order could change between Python versions, so we might
+        # choose to remove this test, or adapt it in the future.
         self.assert_poll_call_functions(
             output,
-            ["a", "b", "c", "d", "e", "f", "g", "h"],
+            ["d", "h", "e", "f", "g", "a", "b", "c"],
```

Comment:
I considered it but I thought it would be better to keep it simple and be notified if the order ever changes, sorting could mask other unintended changes, let's keep this for when we know we have a problem, so far the order is the same from 3.8 to 3.12.